homeassistant/components/noaa_tides/__init__.py

```diff
@@ -1 +1,81 @@
 """The noaa_tides component."""
+
+from dataclasses import dataclass
+
+from noaa_coops.station import Station
+from requests.exceptions import ConnectionError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_TIME_ZONE, CONF_UNIT_SYSTEM, Platform
+from homeassistant.core import _LOGGER, HomeAssistant
+from homeassistant.exceptions import PlatformNotReady
+
+from .const import CONF_STATION_ID, DEFAULT_TIMEZONE, NAME
+from .errors import StationNotFound
+from .helpers import get_default_unit_system
+
+PLATFORMS = [Platform.SENSOR]
+
+type NoaaTidesConfigEntry = ConfigEntry[NoaaTidesConfigEntryData]
+
+
+@dataclass
+class NoaaTidesConfigEntryData:
+    """NOAA Tides data type."""
+
+    station_id: str
+    name: str
+    timezone: str
+    unit_system: str
+    station: Station
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, config_entry: NoaaTidesConfigEntry
+) -> bool:
+    """Set up entry."""
+    station_id = config_entry.data.get(CONF_STATION_ID)
+    name = config_entry.title
+    timezone = config_entry.options.get(CONF_TIME_ZONE, DEFAULT_TIMEZONE)
+    unit_system = config_entry.options.get(
+        CONF_UNIT_SYSTEM, get_default_unit_system(hass)
+    )
+
+    if station_id is None:
+        _LOGGER.error("Station ID is required")
+        raise StationNotFound
+
+    try:
+        station = await hass.async_add_executor_job(Station, station_id, unit_system)
+    except KeyError as exception:
+        _LOGGER.error("%s sensor station_id %s does not exist", NAME, station_id)
+        raise StationNotFound from exception
+    except ConnectionError as exception:
+        _LOGGER.error(
+            "Connection error during setup in %s sensor for station_id: %s",
+            NAME,
+            station_id,
+        )
+        raise PlatformNotReady from exception
```

Comment:
we can also put the error messages as the parameter for the errors, this way they show up on the frontend